### PR TITLE
Improve categorical converter error message

### PR DESCRIPTION
**Pull request by dstansby**
## PR Summary
xref #11095. Provide a nicer error message if the user tries to set a `unit` without `_mapping` to a categorical converter.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/next_api_changes/* if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by QuLogic** [comment]
It's slightly more specific than an `AttributeError`, but I'm not sure what the user is supposed to do with this information?

**Comment by anntzer** [comment]
I agree the error message should be something more actionable e.g. "not a valid categorical unit" or similar

**Comment by dstansby** [comment]
How about now? I think this is a small improvement avoids the user having to work out where and why the attribute error was thrown, immediately saying "the problem was with the unit passed to a categorical converter".

File: lib/matplotlib/category.py [hunk 1]
```diff
@@ -109,6 +111,13 @@ def default_units(data, axis):
             axis.units.update(data)
         return axis.units
 
+    @staticmethod
+    def _validate_unit(unit):
+        if not hasattr(unit, '_mapping'):
+            raise ValueError(
+                f'Provided unit "{unit}" is not a valid for a categorical '
```
Comment (QuLogic) [inline lib/matplotlib/category.py:118]: 'a valid' should remove 'a', or add another noun after 'valid'

**Comment by tacaswell** [comment]
It is at least a less-confusing error message that the type error (for the same private attribute) that they are getting now.

**Pull request merged**